graylog2-server/src/main/java/org/graylog2/users/UserImpl.java

```diff
@@ -270,20 +270,21 @@ public DateTimeZone getTimeZone() {
 
     @Override
     public void setTimeZone(final String timeZone) {
-        DateTimeZone dateTimeZone;
-        try {
-            dateTimeZone = DateTimeZone.forID(firstNonNull(timeZone, DateTimeZone.UTC.getID()));
-        } catch (IllegalArgumentException e) {
-            LOG.info("Invalid timezone \"{}\", falling back to UTC.", timeZone);
-            dateTimeZone = DateTimeZone.UTC;
+        DateTimeZone dateTimeZone = null;
+        if (timeZone != null) {
+            try {
+                dateTimeZone = DateTimeZone.forID(firstNonNull(timeZone, DateTimeZone.UTC.getID()));
+            } catch (IllegalArgumentException e) {
+                LOG.info("Invalid timezone \"{}\", falling back to UTC.", timeZone);
+                dateTimeZone = DateTimeZone.UTC;
```

Comment:
I do not think we should do this. Using an incorrect timezone name indicates a programming error/misuse of the API, so we should handle this accordingly and return an error to the caller without changing the timezone of the user. If we leave it that way, the caller has no way to find out from the result of the call if the timezone was set properly. The message logged is also not helpful, it is only visible to the server admin and does not even include the user name.

I know this is the way the code was before, but if you are at it, I would be glad if you could improve that part too :)